carla/models/catalog/catalog.py

```diff
@@ -0,0 +1,154 @@
+import numpy as np
+
+from ..api import MLModel
+from .load_model import load_model
+
+
+class MLModelCatalog(MLModel):
+    def __init__(
+        self, data_name, model_type, ext="h5", cache=True, models_home=None, **kws
+    ):
+        self._data_name = data_name
+        self._model = load_model(model_type, data_name, ext, cache, models_home, **kws)
+
+        if ext == "pt":
+            self._backend = "pytorch"
+        elif ext == "h5":
+            self._backend = "tensorflow"
+        else:
+            raise Exception("Model type not in catalog")
+
+        self._name = model_type + "_" + data_name
+
+        if data_name == "adult":
+            self._feature_input_order = [
+                "age",
+                "fnlwgt",
+                "education-num",
+                "capital-gain",
+                "capital-loss",
+                "hours-per-week",
+                "workclass_Private",
+                "marital-status_Non-Married",
+                "occupation_Other",
+                "relationship_Non-Husband",
+                "race_White",
+                "sex_Male",
+                "native-country_US",
+            ]
+        else:
+            raise Exception("Model for dataset not in catalog")
+
+    @property
+    def name(self):
+        """
+        Contains meta information about the model, like name, dataset, structure, etc.
+
+        Returns
+        -------
+        name : String
+            Individual name of the ml model
+        """
+        return self._name
+
+    @property
+    def feature_input_order(self):
+        """
+        Saves the required order of feature as list.
+
+        Prevents confusion about correct order of input features in evaluation
+
+        Returns
+        -------
+        ordered_features : List of String
+            Correct order of input features for ml model
+        """
+        return self._feature_input_order
+
+    @property
+    def backend(self):
+        """
+        Describes the type of backend which is used for the ml model.
+
+        E.g., tensorflow, pytorch, sklearn, ...
+
+        Returns
+        -------
+        backend : String
+            Used framework
+        """
+        return self._backend
+
+    @property
+    def raw_model(self):
```

Comment:
Should we just name it `model`? I think `MLCatalog.model` would be pretty clear

Reply:
Thought raw_model would be fitting with the data.raw_data structure, but model would be fine with me :)

Reply:
Also true, for me personally raw_model has less meaning then raw_data. But no hard opinions, feel free to do either :)